### ISSUE TITLE
Random crash

### ISSUE THREAD

**Issue by falzm** [justjanne/powerline-go#22]
Hi

Every once in a while, the `powerline-go` process crashes at startup:

```
failed MSpanList_Insert 0x2b3000 0x1f8405c5e7e0 0x0 0x0
fatal error: MSpanList_Insert

runtime stack:
runtime.throw(0x1a8cb0, 0x10)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/panic.go:530 +0x90 fp=0x7fff5fbfee50 sp=0x7fff5fbfee38
runtime.(*mSpanList).insert(0x265fc8, 0x2b3000)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mheap.go:933 +0x293 fp=0x7fff5fbfee80 sp=0x7fff5fbfee50
runtime.(*mheap).freeSpanLocked(0x2657c0, 0x2b3000, 0x100, 0x0)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mheap.go:809 +0x4be fp=0x7fff5fbfeee8 sp=0x7fff5fbfee80
runtime.(*mheap).grow(0x2657c0, 0x8, 0x0)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mheap.go:675 +0x2a0 fp=0x7fff5fbfef40 sp=0x7fff5fbfeee8
runtime.(*mheap).allocSpanLocked(0x2657c0, 0x1, 0x0)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mheap.go:553 +0x4e3 fp=0x7fff5fbfef98 sp=0x7fff5fbfef40
runtime.(*mheap).alloc_m(0x2657c0, 0x1, 0x15, 0x0)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mheap.go:437 +0x119 fp=0x7fff5fbfefc8 sp=0x7fff5fbfef98
runtime.(*mheap).alloc.func1()
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mheap.go:502 +0x41 fp=0x7fff5fbfeff8 sp=0x7fff5fbfefc8
runtime.systemstack(0x7fff5fbff018)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/asm_amd64.s:307 +0xab fp=0x7fff5fbff000 sp=0x7fff5fbfeff8
runtime.(*mheap).alloc(0x2657c0, 0x1, 0x10000000015, 0x1b8df)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mheap.go:503 +0x63 fp=0x7fff5fbff048 sp=0x7fff5fbff000
runtime.(*mcentral).grow(0x2673c0, 0x0)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mcentral.go:209 +0x93 fp=0x7fff5fbff0b0 sp=0x7fff5fbff048
runtime.(*mcentral).cacheSpan(0x2673c0, 0x262378)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mcentral.go:89 +0x47d fp=0x7fff5fbff0f0 sp=0x7fff5fbff0b0
runtime.(*mcache).refill(0x2af000, 0x15, 0x7fff5fbff158)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/mcache.go:119 +0xcc fp=0x7fff5fbff128 sp=0x7fff5fbff0f0
runtime.mallocgc.func2()
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/malloc.go:642 +0x2b fp=0x7fff5fbff148 sp=0x7fff5fbff128
runtime.systemstack(0x7fff5fbff1e8)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/asm_amd64.s:307 +0xab fp=0x7fff5fbff150 sp=0x7fff5fbff148
runtime.mallocgc(0x180, 0x185e00, 0x0, 0x800000000)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/malloc.go:643 +0x869 fp=0x7fff5fbff228 sp=0x7fff5fbff150
runtime.newobject(0x185e00, 0x262690)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/malloc.go:781 +0x42 fp=0x7fff5fbff250 sp=0x7fff5fbff228
runtime.malg(0x8000, 0x262900)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/proc.go:2634 +0x27 fp=0x7fff5fbff288 sp=0x7fff5fbff250
runtime.mpreinit(0x262c40)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/os1_darwin.go:140 +0x1f fp=0x7fff5fbff2a0 sp=0x7fff5fbff288
runtime.mcommoninit(0x262c40)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/proc.go:494 +0x105 fp=0x7fff5fbff2e8 sp=0x7fff5fbff2a0
runtime.schedinit()
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/proc.go:434 +0x79 fp=0x7fff5fbff330 sp=0x7fff5fbff2e8
runtime.rt0_go(0x7fff5fbff360, 0x3, 0x7fff5fbff360, 0x0, 0x3, 0x7fff5fbff530, 0x7fff5fbff54d, 0x7fff5fbff560, 0x0, 0x7fff5fbff562, ...)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/asm_amd64.s:138 +0x132 fp=0x7fff5fbff338 sp=0x7fff5fbff330
```

It appears to be quite random, sorry for not being able to provide more reproduction conditions. I run it on macOS X (darwin-amd64) with `-colorize-hostname` option. Let me know if I can provide more information to help.

**Comment by justjanne** [comment]
This seems to be an issue with go itself, not with this project. This crash happens before any of our code even started to run, which is very unusual.

I can’t really do anything about it, sadly.

**Comment by cardonator** [comment]
This identical error happens to other people, including the folder and go version on the `releases` executable. In one instance I've seen it was on a Mac with go 1.9 installed. This seems to have something to do with the version of go used in travis. But it's not apparent how that plays into this package.

**Comment by cardonator** [comment]
I see, your .travis.yml specifies go 1.6 in it.

https://github.com/justjanne/powerline-go/blob/master/.travis.yml

and I'm guessing you use travis for CI to build your releases. Which is fine but that's definitely where the issue stems from.

**Comment by justjanne** [comment]
If you build with Go 1.9 yourself, does it work fine? Because I’ll just upgrade the version in the travis now, but would prefer if that would definitely fix this.

**Comment by cardonator** [comment]
We can try to build on OSX on Friday with go1.9 to find out.

**Comment by cardonator** [comment]
We were successfully able to build on 1.9, so now we are going to see if we get any crash over the next few minutes.

**Comment by cardonator** [comment]
Okay, we've been trying for the past half hour. Haven't seen a single crash. It's a big change from the release version.

**Comment by cardonator** [comment]
We've been using this for a week now and the crashes seem to have subsided.

**Comment by falzm** [comment]
@cardonator by "subsided" do you mean that you haven't had any more crashes, or the frequency reduced?

**Comment by cardonator** [comment]
Haha, good point. The crashes have completely gone away after building on go1.9.

**Comment by cardonator** [comment]
It looks like this should be fixed in v1.6.0 with commit edb37971c01190e1c834514df71e8951084f6b30

@falzm can you confirm?

**Comment by falzm** [comment]
@cardonator I confirm that I haven't seen a crash since I've upgraded to 1.6.0. I think it's OK to close this issue now, and possibly re-open it if it happens again.

Thank you for fixing this 😊
